### ISSUE TITLE
ListOfType eagerly loads types during construction when use in combination with NonNull

### ISSUE THREAD

**Issue by le0daniel** [webonyx/graphql-php#1268]
**Problem**:
If you use ListOf and NonNull in combination, lazy loading is not working as expected. It always eagerly loads the the types. The issue is really in how the assertion description is computed ahead of time, before the assertion failed. 

1. The constructor of ListOfType calls `Type::assertType($type)` during construction.
2. `Type::assertType($type)` creates a new message `new InvariantViolation('Expected ' . Utils::printSafe($type) . ' to be a GraphQL type.')`.
3. ` Utils::printSafe($type)` eagerly loads wrapped types.

In short, if you use `Type::listOfType(Type::nonNull(fn() => 'resolveLazyType'))`, this will eagerly load the type.

**Question**:
Why is the ListOfType doing this assertion here? Is this intentional?

**Comment by spawnia** [comment]
This will be fixed in the upcoming v15.
